src/renderers/dom/client/eventPlugins/BeforeInputEventPlugin.js

```diff
@@ -457,20 +457,21 @@ var BeforeInputEventPlugin = {
     nativeEvent,
     nativeEventTarget
   ) {
-    return [
-      extractCompositionEvent(
-        topLevelType,
-        targetInst,
-        nativeEvent,
-        nativeEventTarget
-      ),
-      extractBeforeInputEvent(
-        topLevelType,
-        targetInst,
-        nativeEvent,
-        nativeEventTarget
-      ),
-    ];
+    var compose = extractCompositionEvent(
+      topLevelType,
+      targetInst,
+      nativeEvent,
+      nativeEventTarget
+    );
+
+    var before = extractBeforeInputEvent(
+      topLevelType,
+      targetInst,
+      nativeEvent,
+      nativeEventTarget
+    );
+
+    return compose && before ? [compose, before] : compose ? compose : before;
```

Comment:
`[null, null]` returns for just about every single event plugin test that executes the `EventPluginHub.extractEvents` code path. 

Quickly scanning (throwing on `[null, null]`), this is 22 of 24 SimpleEventPlugin tests, 7 of 9 ChangeEvent plugin tests, and a handful of other event system tests.